samtranslator/model/eventsources/push.py

```diff
@@ -562,6 +562,16 @@ def _add_swagger_integration(self, api, function):
                             'is only a valid value when a DefaultAuthorizer on the API is specified.'.format(
                                 method=self.Method, path=self.Path))
 
+            apikey_required_setting = self.Auth.get('ApiKeyRequired')
+            apikey_required_setting_is_false = apikey_required_setting is not None and not apikey_required_setting
+            if apikey_required_setting_is_false and not api_auth.get('ApiKeyRequired'):
```

Comment:
Throwing an exception here is not required, as this would be a no-op on api gw.

Reply:
Like the Authorizers implementation, I am storing the explicit ApiKeyRequired: False for a function in the security block for that function as `api_key_false`. 
```
"security": { "api_key_false": [] }
```
This security option gets cleaned up by the swagger.set_path_default_apikey_required method, however, this is only called when there is a default ApiKeyRequired setting on the Serverless API.  If there is no default setting (or no explicit Serverless API) then we never loop back and cleanup the `api_key_false` security setting.

I assumed we didn't want to spit out this security setting, however, I'll defer to your judgement.  I'm also open to other ideas on this.